### PR TITLE
rebase: using starbase

### PR DESCRIPTION
**Pull request by syu-w**
CRAFT-1508
CRAFT-1927

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,17 +45,15 @@ on `craft-store-test-charm`, some tests will fail rather than being skipped.
...
+
 
 ## Verifying documentation changes
 
 To locally verify documentation changes run:
 
-```
-make docs
-```
+TODO
```
Comment (lengau) [inline README.md:55]:
```suggestion
`tox run -e lint-docs,build-docs
```

File: craft_store/auth.py [hunk 2]
```diff
@@ -170,7 +176,7 @@ def get_credentials(self) -> str:
             encoded_credentials_string = self._keyring.get_password(
                 self.application_name, self.host
             )
-        except Exception as unknown_error:
+        except errors.CraftStoreError as unknown_error:
```
Comment (lengau) [inline craft_store/auth.py:179]: Please return this to `Exception` and put an appropriate linter silencing comment here. `keyring` won't raise store errors.

File: pyproject.toml [hunk 3]
```diff
@@ -6,32 +106,152 @@ use_parentheses = true
 ensure_newline_before_comments = true
 line_length = 88
 
-[tool.pylint.messages_control]
```
Comment (lengau) [inline pyproject.toml:9]: Since starbase doesn't have pylint, please keep these pylint settings around.

File: pyproject.toml [hunk 4]
```diff
@@ -6,32 +106,152 @@ use_parentheses = true
...
-load-plugins = "pylint_fixme_info,pylint_pytest"
+
+[tool.pyright]
+strict = ["craft_store"]
+pythonVersion = "3.8"
+pythonPlatform = "Linux"
 
 [tool.mypy]
 python_version = "3.8"
-plugins = ["pydantic.mypy"]
```
Comment (lengau) [inline pyproject.toml:27]: Please keep these settings too

File: pyproject.toml (hunk 3 above)
Comment (syu-w) [reply pyproject.toml:9]: But we don't use pylint?

Comment (syu-w) [reply pyproject.toml:9]: pylint added

**Review by lengau — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]

**Pull request merged**